### PR TITLE
Add `--disable-admin-operations` flag in Compactor UI and Bucket UI

### LINKED ISSUE #5390: Provide option to disable block mark deletion and mark no compaction from bucket UI
In Prometheus, admin level functionalities like delete series are not provided through UI.
But in Thanos, now it is supported to mark deletion and mark no compaction from bucket UI. This is an admin-level operation and requires admin permissions. Some users might not want to expose those functionalities from bucket UI to their users.

We can have an option to disable those functionalities from UI or make those features not enabled by default.

### PR DESCRIPTION
**Pull request by harsh-ps-2003**
Signed-off-by: Harsh Pratap Singh <harshpratapsingh8210@gmail.com>
Fixes #5390 

* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

This PR adds a flag, named `--disable-admin-operations`, to the Compactor UI and Bucket Web UI to decide whether to disable the corresponding buttons Mark Deletion and Mark No Compaction in the UI as they are admin-level operations. It also restricts the API usage for both of them.

## Verification

It was tested locally.
The UI test:
Screenshot 2023-08-10 at 9 09 08 PM
Screenshot 2023-08-10 at 9 09 20 PM
Screenshot 2023-08-10 at 9 08 42 PM
Screenshot 2023-08-10 at 9 08 52 PM

The API test:
```
❯ curl -X POST -d "id=_ulid&action=DELETION" http://localhost:8080/api/v1/blocks/mark
{"status":"error","errorType":"bad_data","error":"Admin operations are disabled"}
❯ curl -X POST -d "id=_ulid&action=DELETION" http://localhost:10922/api/v1/blocks/mark
{"status":"error","errorType":"bad_data","error":"Admin operations are disabled"}
```
cc @yeya24 @GiedriusS @saswatamcode

### CODE REVIEW

**Review by yeya24 — APPROVED** [review]

**Comment by harsh-ps-2003** [comment]
The E2E tests are flaky!

File: cmd/thanos/compact.go [hunk 1]
```diff
@@ -786,4 +787,6 @@ func (cc *compactConfig) registerFlag(cmd extkingpin.FlagClause) {
 	cc.webConf.registerFlag(cmd)
 
 	cmd.Flag("bucket-web-label", "External block label to use as group title in the bucket web UI").StringVar(&cc.label)
+
+	cmd.Flag("disable-admin-operations", "Disable UI/API admin operations like marking blocks for deletion and no compaction.").Default("false").BoolVar(&cc.disableAdminOperations)
```
Comment (MichaHoffmann) [inline cmd/thanos/compact.go:791]: can we call it "web.disable-admin-operations" maybe?

Comment (harsh-ps-2003) [reply cmd/thanos/compact.go:791]: I was actaully told by @saswatamcode to make it this way. Still I can ofcourse change the flag name, but still how will the PR get merged when these flaky e2e tests don't pass!

Comment (MichaHoffmann) [reply cmd/thanos/compact.go:791]: We could merge without them passing if they are too flaky and make them stable in another PR

Comment (harsh-ps-2003) [reply cmd/thanos/compact.go:791]: > can we call it "web.disable-admin-operations" maybe?

btw calling it web.disable-admin-operations doesn't seem to be right as it also disables use of API. and the checks are passing now!! i can get a merge maybe

Comment (saswatamcode) [reply cmd/thanos/compact.go:791]: Yup, API & UI both get disabled by the flag, so web prefix is probably not needed. 🙂 

> but still how will the PR get merged when these flaky e2e tests don't pass!

No need to worry so much, there are a few common flakes, which pass with 1/2 retries. I think there are issues to address them.

**Review by saswatamcode — APPROVED** [review]
LGTM!

**Pull request merged**